docs/connection/connection-spans.md

```diff
@@ -0,0 +1,180 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: Connection Spans
+--->
+
+# Semantic Conventions for Connection Spans
+
+This document defines semantic conventions to apply when instrumenting client side of socket connections with spans.
+
+**Status**: [Experimental][DocumentStatus]
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+* [Span name](#span-name)
+* [Attributes](#attributes)
+* [Examples](#examples)
+  * [Successful connection](#successful-connection)
+  * [Successful connect, but connection terminates with an error](#successful-connect-but-connection-terminates-with-an-error)
+  * [Attempt to establish connection ends with `econnrefused` error](#attempt-to-establish-connection-ends-with-econnrefused-error)
+  * [Relationship with application protocols such as HTTP](#relationship-with-application-protocols-such-as-http)
+  * [Connection retry example](#connection-retry-example)
+
+<!-- tocstop -->
+
+this convention defines two types of spans:
+
+- `connect` span: describes the process of establishing a connection. It corresponds to `connect` function ([Linux or other POSIX systems](https://man7.org/linux/man-pages/man2/connect.2.html) /
+[Windows](https://docs.microsoft.com/windows/win32/api/winsock2/nf-winsock2-connect)).
+- `connection` span: describes the connection lifetime: it starts right after the connection is successfully established and ends when connection terminates.
```

Comment:
Not sure if we need both, connect and connection, or if all the data can be represented in the connection.
In an HTTP case, the equivalent of connect is a `wire-request` - as the typical `http request` span tracks a logical operation rather than what happens on the wire. If there is auto-redirection for example as part of the http library, then the `http request` may actually result in multiple `wire-requests` as it retrieves the redirect and then makes a subsequent call for the data.

Reply:
it's important to know how long it takes to establish a connection and important to know if connection was ever established and then terminated.

We can potentially have one span for connection and then indicate when the connection has happened with an event, but I'd still argue that we need two separate metrics.